crates/router/src/core/payments.rs

```diff
@@ -1479,6 +1479,13 @@ where
                 let is_error_in_response = router_data.response.is_err();
                 // If is_error_in_response is true, should_continue_payment should be false, we should throw the error
                 (router_data, !is_error_in_response)
+            } else if connector.connector_name == router_types::Connector::Nmi
+                && !matches!(format!("{operation:?}").as_str(), "CompleteAuthorize")
```

Comment:
Only in case of Payment Confirm we need to trigger Preprocessing

Reply:
`preprocessing_steps` is only implemented for authorize and complete auth flow meaning only these two flow can execute preprocessing. if it matches complete auth flow, it wont enter the body since it is a not operation, leaving only auth flow to enter this body.